router/session.py

```diff
@@ -0,0 +1,103 @@
+from fastapi import APIRouter, HTTPException
+import requests
+from settings import settings
+from datetime import datetime
+
+router = APIRouter(prefix="/session", tags=["Session"])
+session_db_url = settings.db_url + "session/"
+session_start_buffer_time = 900000
+
+def get_current_datetime():
+    """
+    Returns current datetime
+    """
+    return datetime.today()
+
+def get_current_date():
+    """
+    Returns current date
+    """
+    return get_current_datetime().date()
+
+def get_current_timestamp():
+    """
+    Returns current timestamp
+    """
+    return get_current_datetime().timestamp()
+
+def get_date(datetime: datetime):
+    """
+    Returns date for the given datetime
+    """
+    return datetime.date()
+
+def get_timestamp(datetime: datetime):
+    """
+    Returns timestamp for the given datetime
+    """
+    return datetime.timestamp()
+
+def is_start_time_valid(start_time: str, repeat_schedule: str):
+    """
+    Checks if session start time is valid
+    - If session start date is less than or equal to current date:
+        - If session is not repeating, checks if the session start timestamp is less than the current timestamp
+        - If session is repeating, builds a timestamp using current date and session start time and checks that with the current timestamp
+    - Otherwise, returns False
+    """
+    parsed_start_time = datetime.strptime(start_time,'%Y-%m-%dT%H:%M:%SZ')
+    session_start_date, session_start_timestamp = get_date(parsed_start_time), get_timestamp(parsed_start_time)
+    current_date, current_timestamp = get_current_date(), get_current_timestamp()
+    repeated_session_start_datetime = datetime(current_date.year, current_date.month, current_date.day, parsed_start_time.hour, parsed_start_time.minute, parsed_start_time.second)
+    if session_start_date <= current_date:
```

Comment:
If we did not have a repeat schedule, that could be done. But because of the schedule, I'll have to use the time only to build a different datetime to compare with.